Preparing 3.55.0 cumulusci release (#3128)
SFDO-Tooling/CumulusCI: HISTORY.rst

History
=======

3.55.0 (2022-03-24)
-------------------

Changes

> - Tweaked Oauth App org restriction docs. Alters the trigger for warn_oauth_restricted error handler

Reviewer: Docs changes shouldn't go in release notes.

With respect to the other part of the PR: what is the user-facing change? I don't know the relevance of  the `warn_oauth_restricted` trigger error handler to me as an end-user.

to reflect upstream changes. Also adds a note to the onboarding docs to make the workaround
explicit. (#3044)

> - Usage examples for task reference, Adds example usage for a couple of tasks (#3110).

Reviewer 2: ```suggestion
```

> - CumulusCI now provides an option to verify server certificates using root CA certs loaded from the system, instead of the default set of CA certs bundled with the Python requests library. This is considered experimental, so must be opted in by setting the CUMULUSCI_SYSTEM_CERTS environment variable to "True" (#3114)

Reviewer 2: ```suggestion
- CumulusCI now provides an option to verify server certificates using root CA certs loaded from the system, instead of the default set of CA certs bundled with the Python requests library. This is considered experimental, so must be opted in by setting the ``CUMULUSCI_SYSTEM_CERTS`` environment variable to ``True`` (#3114)
```

> - Improvements to the custom locator strategy used by the robot framework keyword input form data to find form fields and lightning components based on label text. This locator strategy (label:) is now up to 10x faster at finding non-lightning form components, and is able to find elements with more complicated labels (eg: labels with text inside of nested spans). (#3117)

Reviewer 2: ```suggestion
- Improvements to the custom locator strategy used by the Robot Framework keyword ``input form data`` to find form fields and lightning components based on label text. This locator strategy (``label:``) is now up to 10x faster at finding non-lightning form components, and is able to find elements with more complicated labels (eg: labels with text inside of nested spans). (#3117)
```

> - Fixed a bug that prevented the use of aliases for robot framework page objects. Improved the documentation for using robot framework page objects and the pageobject decorator. (#3120)

Reviewer 2: ```suggestion
- Fixed a bug that prevented the use of aliases for Robot Framework page objects.(#3120)
```

But this belongs under Issues Closed, not Changes

> - The robot framework keyword 'input form data' now does a better job of working with non-lightning checkboxes. (#3122)

Reviewer: Please review for consistency of how we refer to code keywords like 'input form data', CUMULUSCI_SYSTEM_CERTS, sources, drop_missing_schema. You can look through the rest of the file to see the right syntax.

> - The robot framework keyword 'input form data' now does a better job of working with non-lightning checkboxes. (#3122)

Reviewer 2: ```suggestion
- The Robot Framework keyword ``input form data`` now does a better job of working with non-lightning checkboxes. (#3122)
```

Issues Closed

- Fixed an error that could occur when loading a project that uses very old flow syntax with tasks instead of steps. (#3118)

> - Fixed a bug when using `sources` to access another project from a github branch, where the branch context was not available to branch-based dependency resolvers. (#3123)

Reviewer 2: ```suggestion
- Fixed a bug when using ``sources`` to access another project from a github branch, where the branch context was not available to branch-based dependency resolvers. (#3123)
```

> - The drop_missing_schema option was not respected by the snowfakery task. Now it is. (#3125)

Reviewer 2: ```suggestion
- The ``drop_missing_schema`` option was not respected by the ``snowfakery`` task. Now it is. (#3125)
```
